### PR TITLE
ui: Use CIDs for visiting perforce files from changelist view page

### PR DESCRIPTION
**Pull request by indradhanush**
Stacked on #53605. 

Part of #40330.

## Test plan

🎥 https://www.loom.com/share/a516e86ffc4341f7a7bfd83c1808c7bd?sid=d6d1d5fe-5842-458e-8078-4fb3d4e9d0c1

### CODE REVIEW

File: client/web/src/repo/backend.ts [hunk 1]
```diff
@@ -128,21 +142,24 @@ export const resolveRepoRevision = memoizeObservable(
                 if (!data.repositoryRedirect.mirrorInfo.cloned) {
                     throw new CloneInProgressError(repoName, 'queued for cloning')
                 }
-                if (!data.repositoryRedirect.commit) {
+
+                // The ""revision" we queried for could be a commit or a changelist.
```
Comment (sashaostrikov) [inline client/web/src/repo/backend.ts:146]:
```suggestion
                // The "revision" we queried for could be a commit or a changelist.
```

**Review by peterguy — APPROVED** [review]
worked great locally; LGTM

**Review by sashaostrikov — APPROVED** [review]
:shipit:

**Comment by indradhanush** [comment]
Rebased off main and dropped backend commits from this PR.

**Pull request merged**